trl/trainer/ppo_trainer.py

```diff
@@ -136,10 +136,16 @@ def __init__(
         if data_collator is None:
             data_collator = DataCollatorWithPadding(self.processing_class)
 
-        self.model.generation_config.eos_token_id = (
-            None  # disable `pad_token_id` and `eos_token_id` because we just want to
-        )
-        self.model.generation_config.pad_token_id = None  # generate tokens without truncation / padding
+        # Configure generation settings based on config
+        if hasattr(self.processing_class, "eos_token_id") and self.args.use_model_eos_token:
+            # Use the tokenizer's EOS token if available
+            self.model.generation_config.eos_token_id = self.processing_class.eos_token_id
+        elif hasattr(self.model.config, "eos_token_id") and self.args.use_model_eos_token:
+            # Fallback to model config's EOS token
+            self.model.generation_config.eos_token_id = self.model.config.eos_token_id
```

Comment:
Maybe this line is enough? Why would we need two options?

Reply:
I remove this line 
```python 
elif hasattr(self.model.config, "eos_token_id") and self.args.use_model_eos_token:
    # Fallback to model config's EOS token
    self.model.generation_config.eos_token_id = self.model.config.eos_token_id
```
and kept the one which uses the tokenizer 

I tested it locally and it seems to be doing find the training was done perfectly with and without the eos_token_id
Do i need to like write custom tests or documentation ?